doc/sphinx/Pacemaker_Explained/status.rst

```diff
@@ -1,280 +1,373 @@
-   |                  | values: ``online``, ``offline``.                         |
-   +------------------+----------------------------------------------------------+
-   | crm-debug-origin | .. index::                                               |
-   |                  |    single: crm-debug-origin; node status                 |
-   |                  |    single: node; status, crm-debug-origin                |
-   |                  |                                                          |
-   |                  | The name of the source function that made the most       |
-   |                  | recent change (for debugging purposes).                  |
-   +------------------+----------------------------------------------------------+
-   | join             | .. index::                                               |
-   |                  |    single: join; node status                             |
-   |                  |    single: node; status, join                            |
-   |                  |                                                          |
-   |                  | Does the node participate in hosting resources?          |
-   |                  | Allowed values: ``down``, ``pending``, ``member``.       |
-   |                  | ``banned``.                                              |
-   +------------------+----------------------------------------------------------+
-   | expected         | .. index::                                               |
-   |                  |   single: expected; node status                          |
-   |                  |   single: node; status, expected                         |
-   |                  |                                                          |
-   |                  | Expected value for ``join``.                             |
-   +------------------+----------------------------------------------------------+
-   
-The cluster uses these fields to determine whether, at the node level, the
-node is healthy or is in a failed state and needs to be fenced.
-   
+.. list-table:: **Attributes of a node_state Element**
+   :class: longtable
+   :widths: 1 1 3
+   :header-rows: 1
+
+   * - Name
+     - Type
+     - Description
+   * - .. _node_state_id:
+
+       .. index::
+          pair: node_state; id
+
+       id
+     - :ref:`text <text>`
+     - Node ID (identical to ``id`` of corresponding ``node`` element in the
+       ``configuration`` section)
+   * - .. node_state_uname:
+
+       .. index::
+          pair: node_state; uname
+
+       uname
+     - :ref:`text <text>`
+     - Node name (identical to ``uname`` of corresponding ``node`` element in the
+       ``configuration`` section)
+   * - .. node_state_in_ccm:
+
+       .. index::
+          pair: node_state; in_ccm
+
+       in_ccm
+     - :ref:`epoch time <epoch_time>` *(since 2.1.7; previously boolean)*
+     - If the node's controller is currently in the cluster layer's membership,
```

Comment:
Is it accurate to say the *controller* is in the cluster layer's membership? I know the controller can be in the CPG (which is covered by the `crmd` attribute).

Reply:
Corosync only knows about the node as a whole, but each daemon that uses the cluster layer registers for notifications independently. Since the controller is the one that records node_state, it's the one that matters for this purpose. It could be said more correctly, but not worth any more effort ...